pkg/isb/jetstream/reader.go

```diff
@@ -100,6 +109,67 @@ func (jr *jetStreamReader) Close() error {
 	return nil
 }
 
+func (jr *jetStreamReader) runAckInfomationChecker(ctx context.Context) {
+	js, err := jr.conn.JetStream()
+	if err != nil {
+		// Let it exit if it fails to start the information checker
+		jr.log.Fatal("Failed to get Jet Stream context, %w", err)
+	}
+	checkAckInfo := func() {
+		c, err := js.ConsumerInfo(jr.stream, jr.stream)
+		if err != nil {
+			jr.log.Errorw("failed to get consumer info in the reader", zap.Error(err))
+			return
+		}
+		ts := timestampedSequence{seq: int64(c.AckFloor.Stream), timestamp: time.Now().Unix()}
+		jr.ackedInfo.Append(ts)
+		jr.log.Debugw("Acknowledge information", zap.Int64("ackSeq", ts.seq), zap.Int64("timestamp", ts.timestamp))
+	}
+	checkAckInfo()
+
+	ticker := time.NewTicker(jr.opts.ackInfoCheckInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			checkAckInfo()
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func (jr *jetStreamReader) Pending(_ context.Context) (int64, error) {
+	c, err := jr.js.ConsumerInfo(jr.stream, jr.stream)
+	if err != nil {
+		return isb.PendingNotAvailable, fmt.Errorf("failed to get consumer info, %w", err)
+	}
+	return int64(c.NumPending), nil
+}
+
+// Rate returns the ack rate (tps)
+func (jr *jetStreamReader) Rate(_ context.Context) (float64, error) {
+	if !jr.opts.ackInfoCheck {
```

Comment:
please comment this is for sink. perhaps use `useAckInfoAsSinkRate` or something like this?

Reply:
done.